### ISSUE TITLE
CA1864: false positive / incorrect refactoring

### ISSUE THREAD

**Issue by mortan** [dotnet/roslyn-analyzers#7217]
### Analyzer

**Diagnostic ID**: [CA1864](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1864): `Prefer the 'IDictionary.TryAdd(TKey, TValue)' method`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 8 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.0](https://dotnet.microsoft.com/download/dotnet/8.0)

_OR_

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0

### Describe the bug

Consider the following code which raises a CA1864 warning:
```cs
private Dictionary<string, List<string>> _dictionary = new Dictionary<string, List<string>>();

public void Test(string key)
{
    List<string> data = new List<string>();

    if (!_dictionary.ContainsKey(key))
    {
        DbContext context = new DbContext();
        data = context.LoadData(key);
        if (data != null && data.Any())
        {
            _dictionary.Add(key, data);
        }
    }
}

public class DbContext
{
    public List<string> LoadData(string key) => new List<string>();
}
```

The CA1864 refactoring produces the following code:
```cs
public void Test(string key)
{
    List<string> data = new List<string>();

    if (_dictionary.TryAdd(key, data))
    {
        DbContext context = new DbContext();
        data = context.LoadData(key);
        if (data != null && data.Any())
        {
        }
    }
}
```

This is not equivalent. It seems the rule detects the ContainsKey and the Add, but ignores the if around it.

### Steps To Reproduce

Copy paste the code above in a new console application (.NET 8) and set AnalysisLevel to latest-recommended.

### Expected behavior

The code should not raise a warning.

### Actual behavior

The code raises a warning and the refactoring produces non equivalent code.
